lib/pages/chat/events/download_video_widget.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:dio/dio.dart';
+import 'package:fluffychat/pages/chat/events/download_video_state.dart';
+import 'package:fluffychat/pages/chat/events/event_video_player.dart';
+import 'package:fluffychat/pages/chat/events/message_content_style.dart';
+import 'package:fluffychat/presentation/mixins/handle_video_download_mixin.dart';
+import 'package:fluffychat/presentation/mixins/play_video_action_mixin.dart';
+import 'package:fluffychat/utils/localized_exception_extension.dart';
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/utils/twake_snackbar.dart';
+import 'package:fluffychat/widgets/mxc_image.dart';
+import 'package:fluffychat/widgets/twake_components/twake_icon_button.dart';
+import 'package:fluffychat/widgets/video_viewer_style.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_ref_colors.dart';
+import 'package:matrix/matrix.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+class DownloadVideoWidget extends StatefulWidget {
+  final Event event;
+
+  const DownloadVideoWidget({super.key, required this.event});
+
+  @override
+  State<StatefulWidget> createState() => _DownloadVideoWidgetState();
+}
+
+class _DownloadVideoWidgetState extends State<DownloadVideoWidget>
+    with HandleVideoDownloadMixin, PlayVideoActionMixin {
+  final _downloadStateNotifier = ValueNotifier(DownloadVideoState.initial);
+  String? path;
+  final downloadProgressNotifier = ValueNotifier(0.0);
```

Comment:
Don't forget to dispose these when not in use

Reply:
done